Client/Frontend/Notifications/NotificationRootViewController.swift

```diff
@@ -221,13 +223,40 @@ private extension NotificationRootViewController {
         }
     }
 
-    @objc func didFinishSyncing() {
-        guard showingNotification else { return }
-        dispatch_async(dispatch_get_main_queue()) {
-            self.hideStatusNotification()
+    private func syncMessageForNotification(notificationObject: AnyObject?) -> NSAttributedString? {
+        guard let syncDisplayState = notificationObject as? [String: String],
+        let state = syncDisplayState["state"],
+            let message = syncDisplayState["message"] else {
+                return nil
+        }
+        switch state {
+        case "Error":
+            return NSAttributedString(string: message, attributes: [NSForegroundColorAttributeName: UIConstants.TableViewRowErrorTextColor, NSFontAttributeName: notificationView.titleLabel.font])
+        case "Warning":
+            return NSAttributedString(string: message, attributes: [NSForegroundColorAttributeName: UIConstants.TableViewRowWarningTextColor, NSFontAttributeName: notificationView.titleLabel.font])
+        default: return nil
         }
     }
 
+    @objc func didFinishSyncing(notification: NSNotification) {
+        defer {
+            if showingNotification {
+                showNotificationForSync = false
+                dispatch_async(dispatch_get_main_queue()) {
+                    self.hideStatusNotification()
+                }
+            } else if let syncMessage = notificationView.titleLabel.attributedText {
```

Comment:
I think you want this `if` condition to occur first before the `showingNotification` one. If we are still displaying a `Syncing...` notification and we want to show the sync failed one, the existing notification will be dismissed without showing the next one since the show logic is in the `else` block.